File: library/src/main/kotlin/tag/type/certificate/GblSignatureEcdsaP256.kt
package tag.type.certificate

import tag.GblType
import tag.TagHeader
import tag.Tag
import tag.TagWithHeader

data class GblSignatureEcdsaP256(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    override val tagData: ByteArray,
    val r: UByte,
    val s: UByte
): Tag, TagWithHeader {
    override fun copy(): Tag {
        return GblSignatureEcdsaP256(
            tagHeader = tagHeader,
            tagType = tagType,
            tagData = tagData,
            r = r,
            s = s
        )
    }
}

File: python-library/tag/type/certificate/gbl_signature_ecdsa_p256.py
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ...gbl_type import GblType
    from ...tag_header import TagHeader
    from ...tag import Tag


# Імпорти з інших модулів (будуть додані пізніше):
# from ...tag import Tag
# from ...tag_with_header import TagWithHeader


@dataclass
class GblSignatureEcdsaP256:
    """GBL Signature ECDSA P256 tag"""

    tag_header: 'TagHeader'
    tag_type: 'GblType'
    tag_data: bytes
    r: int  # UByte -> int
    s: int  # UByte -> int

    def copy(self) -> 'Tag':
        """Create a copy of the tag"""
        return GblSignatureEcdsaP256(
            tag_header=self.tag_header,
            tag_type=self.tag_type,
            tag_data=self.tag_data,
            r=self.r,
            s=self.s
        )